index.js

```diff
@@ -188,7 +190,7 @@ Compiler.prototype.compileMessage = function(message, root) {
   return res;
 };
 
-module.exports = function(filename, model) {
+module.exports = function(filename, model,allow_additional_props) {
   var compiler = new Compiler(filename);
-  return compiler.compile(model);
+  return compiler.compile(model,allow_additional_props);
```

Comment:
I meant make this one `compile(opts, model)` and make all the other function calls also use `opts` as the first parameter.